code/game/objects/items/devices/radio/radio.dm

```diff
@@ -261,15 +261,16 @@
 
 	// From the channel, determine the frequency and get a reference to it.
 	var/freq
-	if(channel && channels)
-		if(channel == MODE_DEPARTMENT && channels.len > 0)
+	if(channel && length(channels) && length(secure_radio_connections))
+		if(channel == MODE_DEPARTMENT)
 			channel = channels[1]
 		freq = secure_radio_connections[channel]
 		if(istype(M, /mob) && !freq && channel != RADIO_CHANNEL_UPLINK)
 			to_chat(M, "<span class='warning'>You can't access this channel without an encryption key!</span>")
 		if (!channels[channel]) // if the channel is turned off, don't broadcast
 			return
-	else
```

Comment:
This seems like an arbitrary change - a frequency is always set in the if statement above, this is a minor but I'm ecessaty extra calculation every time this runs without the else.

Reply:
It's not really an arbitrary change. You can pass the if check while you can failed to get any value from `secure_radio_connections` list. That's why I made `if(!freq)`